### ISSUE TITLE
Float tokenization requires whitespace after each float

### ISSUE THREAD

**Issue by PKJedi** [CodyJasonBennett/shaderkit#11]
Thank you for making and open sourcing this tool!

I ran into an issue with source code without whitespace between numbers, operators and identifiers:

```js
tokenize('2.2+bar')
// [ { type: 'float', value: '2.2+bar' } ]
```

presumably an issue with
```js
const isNumber = (c: number) => isAlpha(c) || isDigit(c) || c === PLUS || c === MINUS || c === DOT
```
and/or
```js
while (isNumber(code.charCodeAt(index))) value += code[index++]
```

**Comment by CodyJasonBennett** [comment]
Those two lines are indeed the problem. The number matcher tries to match scientific notation, but it will have to narrow to a suffix in the loop for more context.

**Comment by CodyJasonBennett** [comment]
I just left a position which I created this tool for, but I'm able to dedicate more time here to fix and improve the tool. From a glance, this has been broken since 672a9a5dbad5cbc84d03842124edf9ac173c9ce4 which combined any alphanumerical character. Previously I used regular expressions which are more correct, with no special treatment for uint:

https://github.com/CodyJasonBennett/shaderkit/blob/b70ae781e671480bb4812a144d803c2fb343cfbf/src/tokenizer.ts#L56-L58

A quick hack would be to narrow `isAlpha` in `isNumber` above to E or P, but this can still clash so I'm keen on reverting this behavior to begin with.
